Replace the struct template used in BreakLines::nextBreakablePosition with a simpler struct

darinadler opened the pull request:
#### 63cf646f23f915728d273d5af077b3a69f102ca2
<pre>
Replace the struct template used in BreakLines::nextBreakablePosition with a simpler struct
<a href="https://bugs.webkit.org/show_bug.cgi?id=299672">https://bugs.webkit.org/show_bug.cgi?id=299672</a>
<a href="https://rdar.apple.com/161487336">rdar://161487336</a>

Reviewed by Alan Baradlay.

* Source/WebCore/rendering/BreakLines.h:
(WebCore::BreakLines::nextBreakablePosition): Moved the struct into this function, and used it in a more
straightforward way, obviating the need for function members.

Canonical link: <a href="https://commits.webkit.org/300657@main">https://commits.webkit.org/300657@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/89bb349dee21bfb9df384d7bf0a792d5e48cc28d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123351 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43066 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33762 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130065 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75471 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ff860376-9ca8-4193-a678-9ebf8aa3fb00) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125228 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43789 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93763 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62208 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/31ece2f1-4d80-41d0-9b59-f119945a75bb) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126304 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34890 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110363 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74392 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2cf33d06-7d07-492d-b7af-31ef8fd38fe0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33857 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28522 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73580 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104604 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28748 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132780 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50301 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38279 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102255 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50677 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106587 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102108 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25964 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47461 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25685 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/47088 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50156 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55917 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49627 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52977 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51305 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (approved):
"btw BreakLines is exercised only by SVG text layout by now -which is somewhat ironic as SVG text layout does not break lines (it's just part of the line layout flow)"
ignore me, I was thinking of LineBreaker :)

darinadler committed:
Replace the struct template used in BreakLines::nextBreakablePosition with a simpler struct
https://bugs.webkit.org/show_bug.cgi?id=299672
rdar://161487336

Reviewed by Alan Baradlay.

* Source/WebCore/rendering/BreakLines.h:
(WebCore::BreakLines::nextBreakablePosition): Moved the struct into this function, and used it in a more
straightforward way, obviating the need for function members.

Canonical link: https://commits.webkit.org/300657@main
